sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java

```diff
@@ -567,13 +575,7 @@ private RntbdRequestRecord addPendingRequestRecord(final ChannelHandlerContext c
             final Timeout pendingRequestTimeout = record.newTimeout(timeout -> {
 
                 // We don't wish to complete on the timeout thread, but rather on a thread doled out by our executor
```

Comment:
Clarification: Scheduling is to unblock timer-wheel right? 

Thought: Leaving the choice to consumers is error prone, is it possible to force timer wheel it-self own and schedule on them (Of-course follow-up)?

Reply:
Correct